ui/pages/confirmations/components/confirm/info/hooks/useFeeCalculations.ts

```diff
@@ -52,19 +53,36 @@ export function useFeeCalculations(transactionMeta: TransactionMeta) {
         }) || 0
       } ${ticker}`;
 
-      const currentCurrencyFee = fiatFormatter(
-        Number(
-          getValueFromWeiHex({
-            value: hexFee,
-            conversionRate,
-            fromCurrency: EtherDenomination.GWEI,
-            toCurrency: currentCurrency,
-            numberOfDecimals: 2,
-          }),
-        ),
+      const decimalCurrentCurrencyFee = Number(
+        getValueFromWeiHex({
+          value: hexFee,
+          conversionRate,
+          fromCurrency: EtherDenomination.GWEI,
+          toCurrency: currentCurrency,
+          numberOfDecimals: 2,
+        }),
       );
 
-      return { currentCurrencyFee, nativeCurrencyFee };
+      let currentCurrencyFee, currentCurrencyFeeWith18SignificantDigits;
+      if (decimalCurrentCurrencyFee === 0) {
```

Comment:
I thought the issue was in some cases we'd round down to zero, but if it's actually zero, can we still not show that?

Meaning should we be checking inside `getValueFromWeiHex` to distinguish between actually zero and rounded to zero?

Reply:
It's an interesting observation, I don't know if we would want to discern between 0 and < 0.01 given the option, but in practice it's a non issue because there are no 0.00 $ gas fees.

Reply:
I agree it's rare, but are there not some chains that don't have network fees? Plus user operations can have no gas fees due to paymasters. A future concern for now.